### PR TITLE
Update the nextjs default middleware matcher

### PR DESCRIPTION
**Pull request by nikosdouvlis**
### Explanation:
For more context, please refer to https://github.com/clerk/javascript/pull/3741 

### This PR:
- Updates all instances of the old nextjs matcher with the updated one.

### CODE REVIEW

**Review by dimkl — APPROVED** [review]

**Review by anagstef — APPROVED** [review]

File: docs/advanced-usage/satellite-domains.mdx [hunk 1]
```diff
@@ -256,7 +256,21 @@ You can configure your satellite application by setting the following properties
...
+      ],
+    };
+
+    export const config = {
+      matcher: [
+        // Skip Next.js internals and all static files, unless found in search params
+        '/((?!_next|[^?]*\\.(?:html?|css|js(?!on)|jpe?g|webp|png|gif|svg|ttf|woff2?|ico|csv|docx?|xlsx?|zip|webmanifest)).*)',
+        // Always run for API routes
+        '/(api|trpc)(.*)',
+      ],
```
Comment (jescalan) [inline docs/advanced-usage/satellite-domains.mdx:273]: This looks like it is in here 2x, I assume this was an accident?

File: docs/guides/custom-redirects.mdx [hunk 2]
```diff
@@ -71,7 +71,12 @@ export default clerkMiddleware((auth, req) => {
 });
 
 export const config = {
-  matcher: ['/((?!.*\\..*|_next).*)', '/', '/(api|trpc)(.*)'],
+  matcher: [
+    // Skip Next.js internals and all static files, unless found in search params
+    '/((?!_next|[^?]*\\.(?:html?|css|js(?!on)|jpe?g|webp|png|gif|svg|ttf|woff2?|ico|csv|docx?|xlsx?|zip|webmanifest)).*)',
+    // Always run for API routes
+    '/(api|trpc)(.*)',
+  ],
```
Comment (jescalan) [inline docs/guides/custom-redirects.mdx:79]: Do we want to replace with the commented version everywhere? At the moment, we have a commented version explaining the regex in the main docs, but in other places that it's referenced in code examples, we leave out the comments

**Review by jescalan — CHANGES_REQUESTED** [review]
Just a couple small things

File: docs/guides/custom-redirects.mdx (hunk 2 above)
Comment (nikosdouvlis) [reply docs/guides/custom-redirects.mdx:79]: The quickstart guide didn't have comments before and I guess this is where most people learn about the matcher, that's why I included the comments there. Why woudln't we want the comments everywhere? People would probably just copy and paste the matcher when they find it I guess

File: docs/advanced-usage/satellite-domains.mdx (hunk 1 above)
Comment (nikosdouvlis) [reply docs/advanced-usage/satellite-domains.mdx]: Done

File: docs/guides/custom-redirects.mdx (hunk 2 above)
Comment (alexisintech) [reply docs/guides/custom-redirects.mdx:79]: I say let's be consistent and keep the comments everywhere - we never know where exactly in the docs someone might be copying and pasting code from and comments don't hurt

**Review by alexisintech — APPROVED** [review]

**Review by alexisintech — APPROVED** [review]

**Pull request merged**